src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandlerTest.java

```diff
@@ -243,4 +261,69 @@ private void populatePostponementSscsCaseData() {
                         .documentType(POSTPONEMENT_REQUEST_DIRECTION_NOTICE.getValue())
                         .build()).build();
     }
+
+    @Test
+    public void snlEnabled_givenAGrantedPostponementAndReadyToList_shouldSendCancellation() {
+        populatePostponementSscsCaseData();
+
+        sscsCaseData.setSchedulingAndListingFields(SchedulingAndListingFields.builder()
+            .hearingRoute(HearingRoute.LIST_ASSIST).build());
+        sscsCaseData.setPostponementRequest(PostponementRequest.builder().actionPostponementRequestSelected("grant")
+            .listingOption(State.READY_TO_LIST.getId()).build());
+
+        sscsCaseData.setHearings(List.of(Hearing.builder().value(HearingDetails.builder()
+            .hearingDate(LocalDate.now().plusDays(1).toString())
+            .time("10:00")
+            .build()).build()));
+        sscsCaseData.setAppeal(Appeal.builder().hearingOptions(HearingOptions.builder().build()).build());
+
+        PreSubmitCallbackResponse<SscsCaseData> response =
+            handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
```

Comment:
These three tests are all very similar. Could we extract the common code into a helper function to avoid repetition?

Reply:
Your right, I was just using what was already there; I have refined the whole class now.